src/transformers/generation/utils.py

```diff
@@ -3263,6 +3264,8 @@ def beam_search(
             )
             if model_kwargs["past_key_values"] is not None:
                 model_kwargs["past_key_values"] = self._reorder_cache(model_kwargs["past_key_values"], beam_idx)
+                if not model_kwargs.get("use_legacy_cache", True):
```

Comment:
```suggestion
                if not isinstance(past_key_values, Cache):
```
that's better IMO as it means we don't rely on the `use_legacy_cache` model_kwarg in case people forget to add it. The logic of "always convert to a `Cache` object when we don't have one should hold".